Renames "libtool" -> "${TARGET_PREFIX}libtool" which makes sure
it can't be confused with the host libtool.


After discussions with RP Changed host_alias to build_alias 
at one place as it was causing man page building issue while 
cross compiling for different architectures.

Date: 2010/06/28
Signed-Off-By: Nitin A Kamble <nitin.a.kamble@intel.com>


Index: libtool-2.2.10/libltdl/m4/libtool.m4
===================================================================
--- libtool-2.2.10.orig/libltdl/m4/libtool.m4
+++ libtool-2.2.10/libltdl/m4/libtool.m4
@@ -94,7 +94,8 @@ _LT_SET_OPTIONS([$0], [$1])
 LIBTOOL_DEPS="$ltmain"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+LIBTOOL='$(SHELL) $(top_builddir)'
+LIBTOOL="$LIBTOOL/${host_alias}-libtool"
 AC_SUBST(LIBTOOL)dnl
 
 _LT_SETUP
@@ -201,7 +202,7 @@ aix3*)
 esac
 
 # Global variables:
-ofile=libtool
+ofile=${host_alias}-libtool
 can_build_shared=yes
 
 # All known linkers require a `.a' archive for static linking (except MSVC,
Index: libtool-2.2.10/Makefile.am
===================================================================
--- libtool-2.2.10.orig/Makefile.am
+++ libtool-2.2.10/Makefile.am
@@ -31,7 +31,7 @@ AM_LDFLAGS		=
 DIST_SUBDIRS		= .
 EXTRA_DIST		=
 
-BUILT_SOURCES		= libtool libtoolize
+BUILT_SOURCES		= $(host_alias)-libtool libtoolize
 
 CLEANFILES		=
 MOSTLYCLEANFILES	=
@@ -65,7 +65,7 @@ rebuild = rebuild=:; $(timestamp); corre
 ## ---------------- ##
 
 # The libtool distributor and the standalone libtool script.
-bin_SCRIPTS = libtoolize libtool
+bin_SCRIPTS = libtoolize $(host_alias)-libtool
 
 libtoolize: $(srcdir)/libtoolize.in $(top_builddir)/config.status
 	rm -f libtoolize.tmp libtoolize
@@ -91,8 +91,8 @@ $(srcdir)/libtoolize.in: $(sh_files) lib
 # We used to do this with a 'stamp-vcl' file, but non-gmake builds
 # would rerun configure on every invocation, so now we manually
 # check the version numbers from the build rule when necessary.
-libtool: $(top_builddir)/config.status $(srcdir)/$(auxdir)/ltmain.sh ChangeLog
-	@target=libtool; $(rebuild); \
+$(host_alias)-libtool: $(top_builddir)/config.status $(srcdir)/$(auxdir)/ltmain.sh ChangeLog
+	@target=$(host_alias)-libtool; $(rebuild); \
 	if test -f "$$target"; then \
 	  set dummy `./$$target --version | sed 1q`; actualver="$$5"; \
 	  test "$$actualver" = "$$correctver" && rebuild=false; \
@@ -101,8 +101,8 @@ libtool: $(top_builddir)/config.status $
 	  case $$prereq in *ChangeLog);; *) rebuild=:;; esac; \
 	done; \
 	if $$rebuild; then \
-	  echo $(SHELL) ./config.status $$target; \
-	  cd $(top_builddir) && $(SHELL) ./config.status $$target; \
+	  echo $(SHELL) ./config.status libtool; \
+	  cd $(top_builddir) && $(SHELL) ./config.status libtool; \
 	fi
 
 .PHONY: configure-subdirs
@@ -147,7 +147,7 @@ EXTRA_DIST     += bootstrap $(srcdir)/li
 		  ChangeLog.2002 ChangeLog.2003 ChangeLog.2004 \
 		  ChangeLog.2005 ChangeLog.2006 ChangeLog.2007 \
 		  ChangeLog.2008 ChangeLog.2009
-CLEANFILES     += libtool libtoolize libtoolize.tmp \
+CLEANFILES     += $(host_alias)-libtool libtoolize libtoolize.tmp \
 		  $(auxdir)/ltmain.tmp $(m4dir)/ltversion.tmp
 
 ## We build ltversion.m4 here, instead of from config.status,
@@ -337,7 +337,7 @@ update_mans = \
   PATH=.$(PATH_SEPARATOR)$$PATH; export PATH; \
   $(HELP2MAN) --output=$@
 $(srcdir)/doc/libtool.1: $(srcdir)/$(auxdir)/ltmain.sh
-	$(update_mans) --help-option=--help-all libtool
+	$(update_mans) --help-option=--help-all ${build_alias}-libtool
 $(srcdir)/doc/libtoolize.1: $(srcdir)/libtoolize.in
 	$(update_mans) libtoolize
 
@@ -523,12 +523,12 @@ TESTS_ENVIRONMENT = MAKE="$(MAKE)" CC="$
 
 BUILDCHECK_ENVIRONMENT = _lt_pkgdatadir="$(abs_top_srcdir)" \
 	LIBTOOLIZE="$(abs_top_builddir)/libtoolize" \
-	LIBTOOL="$(abs_top_builddir)/libtool" \
+	LIBTOOL="$(abs_top_builddir)/$(host_alias)-libtool" \
 	tst_aclocaldir="$(abs_top_srcdir)/libltdl/m4"
 
 INSTALLCHECK_ENVIRONMENT = \
 	LIBTOOLIZE="$(bindir)/`echo libtoolize | sed '$(program_transform_name)'`" \
-	LIBTOOL="$(bindir)/`echo libtool | sed '$(program_transform_name)'`" \
+	LIBTOOL="$(bindir)/`echo $(host_alias)-libtool | sed '$(program_transform_name)'`" \
 	LTDLINCL="-I$(includedir)" \
 	LIBLTDL="$(libdir)/libltdl.la" \
 	tst_aclocaldir="$(aclocaldir)"
@@ -679,7 +679,7 @@ if HAVE_FC
 TESTS += $(FC_TESTS)
 endif
 
-tests/demo-conf.test: libtool
+tests/demo-conf.test: $(host_alias)-libtool
 
 EXTRA_DIST     += $(srcdir)/tests/defs.in tests/defs.m4sh \
 		  $(COMMON_TESTS) $(CXX_TESTS) $(F77_TESTS) $(FC_TESTS)
